src/main/webapp/resources/js/pages/projects/create/CreateProjectSamples.js

```diff
@@ -0,0 +1,126 @@
+  Input,
+  Space,
+  Table,
+  Typography,
+} from "antd";
+import React from "react";
+import { useGetCartSamplesQuery } from "../../../apis/cart/cart";
+import { IconExperiment } from "../../../components/icons/Icons";
+import { SampleDetailViewer } from "../../../components/samples/SampleDetailViewer";
+import { blue6 } from "../../../styles/colors";
+
+/**
+ * Component to render samples that are in the cart (if any).
+ * User can select which samples to add to the new project.
+ * @param {Object} form - Ant Design form API
+ * @returns {JSX.Element}
+ * @constructor
+ */
+export function CreateProjectSamples({ form }) {
+  const { data: samples = {}, isLoading } = useGetCartSamplesQuery();
+  const [organismFilter, setOrganismFilter] = React.useState([]);
+  const [selected, setSelected] = React.useState([]);
+  const [lock, setLock] = React.useState(false);
+
+  React.useEffect(() => {
+    const exists = {};
+    samples.unlocked?.forEach((sample) => {
+      if (!exists[sample.organism]) {
+        exists[sample.organism] = {
+          text: sample.organism,
+          value: sample.organism,
+        };
+      }
+    });
+    setOrganismFilter(Object.values(exists));
+  }, [samples]);
+
+  return (
+    <>
+      {samples.locked?.length > 0 && (
+        <Alert
+          type="warning"
+          showIcon
+          message={i18n("CreateProjectSamples.locked")}
+        />
+      )}
+      {samples.unlocked?.length ? (
+        <Space style={{ display: "block" }}>
+          <Table
+            className="t-samples"
+            loading={isLoading}
+            rowSelection={{
+              type: "checkbox",
+              selectedRowKeys: selected,
+              onChange: (selectedRowKeys, selectedRows) => {
+                setSelected(selectedRowKeys);
+                form.setFieldsValue({
+                  samples: selectedRows.map((s) => Number(s.identifier)),
+                });
+              },
+            }}
+            scroll={{ y: 600 }}
+            pagination={false}
+            dataSource={samples.unlocked}
+            rowKey={(sample) => `sample-${sample.identifier}`}
+            columns={[
+              {
+                title: i18n("CreateProjectSamples.sampleName"),
+                dataIndex: "label",
+                render: (text, sample) => (
+                  <SampleDetailViewer sampleId={sample.identifier}>
+                    <Button size="small">{sample.label}</Button>
+                  </SampleDetailViewer>
+                ),
+                onFilter: (value, record) =>
+                  record.label.toLowerCase().indexOf(value.toLowerCase()) >= 0,
+              },
+              {
+                title: i18n("CreateProjectSamples.organism"),
+                dataIndex: "organism",
+                filters: organismFilter,
+                onFilter: (value, record) => record.organism === value,
+              },
+            ]}
+          />
+          <Form.Item name="samples" hidden>
```

Comment:
Just a clarification. Why is this input hidden?

Reply:
Because currently the table checkboxes are not part of the form.  To fix this (unless you know another way), when the table is updated, the values from the checkboses get set as the value of this hidden input.

Reply:
Had a brain cramp there. That's the only way I'm aware of how to get that to work. Thanks